src/Features/AutoAnonymize/Functions/AutoClean.ts

```diff
@@ -57,25 +58,33 @@ export const AutoAnon = async (
     tagsToAnon: any[]
 ) => {
     const newFiles: any = [];
+    const { setLoading } = useStore.getState();
+    
+    try {
+        setLoading(true);
+        dicomData.forEach((dicom: any, index: number) => {
+            const formattedData = FormatData(dicom, tagsToAnon);
 
-    dicomData.forEach((dicom: any, index: number) => {
-        const formatedData = FormatData(dicom, tagsToAnon);
+            // Update the formatted data with the new values from anonTags
+            anonTags.forEach((anonTag) => {
+                const tagIndex = formattedData.findIndex(
+                    (tag: any) => tag.tagId === anonTag.tagId
+                );
+                if (tagIndex !== -1) {
+                    formattedData[tagIndex].newValue = anonTag.newValue;
+                }
+            });
 
-        // Update the formatted data with the new values from anonTags
-        anonTags.forEach((anonTag) => {
-            const tagIndex = formatedData.findIndex(
-                (tag: any) => tag.tagId === anonTag.tagId
-            );
-            if (tagIndex !== -1) {
-                formatedData[tagIndex].newValue = anonTag.newValue;
-            }
-        });
-
-        const updatedFile = tagUpdater(dicomData[0].DicomDataSet, formatedData);
+            const updatedFile = tagUpdater(dicomData[0].DicomDataSet, formattedData);
 
-        newFiles.push(createFile(files[index].name, updatedFile, true));
-    });
+            newFiles.push(createFile(files[index].name, updatedFile, true));
+        });
 
-    const zipFile = await createZipFromFiles(newFiles);
-    downloadDicomFile({ name: "updateDicoms.zip", content: zipFile });
+        const zipFile = await createZipFromFiles(newFiles);
+        downloadDicomFile({ name: "updateDicoms.zip", content: zipFile });
+    } catch (err) {
+        console.error("Auto anonymization failed:", err);
```

Comment:
```suggestion
        logger.error("Auto anonymization failed:", err);
```
should this be a log message rather than console?

look in componenet/utils has a n alertHeader popup could maybe be useful here to let the user know. has know bug #241, but probably okay to start using

Reply:
good catch! logger is definitely a better option